packages/core-js/internals/regexp-sticky-helpers.js

```diff
@@ -0,0 +1,42 @@
+'use strict';
+
+var fails = require('./fails');
+var speciesConstructor = require('../internals/species-constructor');
+
+// babel-minify transpiles RegExp('a', 'y') -> /a/y and it causes SyntaxError,
+// so we use an intermediate function.
+function RE(s, f) {
+  return RegExp(s, f);
+}
+
+exports.UNSUPPORTED_Y = fails(function () {
+  // babel-minify transpiles RegExp('a', 'y') -> /a/y and it causes SyntaxError
+  var re = RE('a', 'y');
+  re.lastIndex = 2;
+  return re.exec('abcd') != null;
+});
+
+exports.BROKEN_CARET = fails(function () {
+  // https://bugzilla.mozilla.org/show_bug.cgi?id=773687
+  var re = RE('^r', 'gy');
+  re.lastIndex = 2;
+  return re.exec('str') != null;
+});
+
+exports.createStickyRegExp = function (re, otherFlags) {
+  var C = speciesConstructor(re, RegExp);
+
+  if (C !== RegExp) return new C(re, otherFlags + 'y');
```

Comment:
It will cause calling `RegExp` with `y` flag in tests like [this](https://github.com/zloirock/core-js/blob/698daaadbec4aab0a72d494382a4f6125b1aed08/tests/tests/es.string.split.js#L741).